### DISCUSSION TITLE
Explicit refetch only works when after invalidating

### DISCUSSION THREAD

**Discussion by finnp** [trpc/trpc#3954] [Q&A]
Hey ☺️ 

I'm currently refactoring a project to use `trpc`, but I haven't used `react-query` much before so sometimes I'm not sure if something is trpc or react-query behavior.

I was just surprised by the following. I have an aggressive cache with a high `staleTime` set. But when certain events happen I want to refetch. Usually I do this through using `invalidate`, but in this case I want to wait until the refetch is done until making changres in the UI to show the data.

However I was noticing that this was not actually working:
```js
        await query.refetch();
```

It was only working when doing
```js
        await util.query.invalidate();
        await query.refetch();
```

I thought from reading `react-query` docs that `refetch` should ignore the `staleTime`, since I'm explictly saying I want to update the cache.

Does anyone know why this needs invalidation? Is this on `react-query` side or on `trpc`? Or am I getting something wrong here?

Best,
Finn

**Comment by KATT** [comment]
`.refetch()` works for me but I can't say I'm sure how the cache settings affect it

This is the API reference: https://github.com/TanStack/query/blob/aa2f94420f17d6ea4d47241e31a0322fe265ee27/docs/react/reference/useQuery.md?plain=1#L256-L262

FWIW, I do this so I never have to think about caching: https://trpc.io/docs/useContext#invalidate-full-cache-on-every-mutation
